### ISSUE TITLE
[QUESTION] How to set permanent sort on table?

### ISSUE THREAD

**Issue by niwasmala** [pankod/refine#1429]
Hi. I'm using useTable with initialSorter without syncWithLocation. But when I change page, the initial sort is missing. I see that there are permanentFilter.

In my case, I have a table of products. Each products have index field as the order to display in store frontpage, and seller can reorder their index.

```tsx
const { tableProps, tableQueryResult, searchFormProps, filters } = useTable<IProduct>({
  initialSorter: [{
    field: 'index',
    order: 'asc',
  }],
  metaData: {
    fields: ['name', 'description''],
  },
});
```

When I open the page, it display the first page correctly by sending sort to graphql query.

![image](https://user-images.githubusercontent.com/476869/149287072-a4bf1393-7401-43f3-a36e-0b5260befccb.png)

But when I change to next page, the sort field is empty.

![image](https://user-images.githubusercontent.com/476869/149287344-cebb41f4-7b34-4824-9360-a7b4ae94a185.png)

Is there any way to set the table with permanent sorter?

**Comment by omeraplak** [comment]
Hey @niwasmala ,
Thank you so much for reaching us. Did you set the `defaultSortOrder` property for the column you gave as `initialSorter`?

Here is an example for you;
````
import {
    Table,
    useTable,
    getDefaultSortOrder,
} from "@pankod/refine";

const { tableProps, tableQueryResult, searchFormProps, filters, sorter } = useTable<IProduct>({
  initialSorter: [{
    field: 'index',
    order: 'asc',
  }],
  metaData: {
    fields: ['name', 'description''],
  },
});

 <Table
      {...tableProps}
      rowKey="id"
      onRow={(record) => {
          return {
              onClick: () => {
                  show("orders", record.id);
              },
          };
      }}>
        <Table.Column
            key="index"
            dataIndex="index"
            sorter
            defaultSortOrder={getDefaultSortOrder("id", index)}
        />
</Table>
````

I hope it was useful for you. Please do not hesitate to contact us if you have any questions.

Best,
Omer | Your friends @pankod

**Comment by niwasmala** [comment]
Hi @omeraplak 

I got an error on `ReferenceError: index is not defined` from `defaultSortOrder={getDefaultSortOrder("id", index)}`.
I change to `defaultSortOrder={getDefaultSortOrder("index", sorter)}` and it works!

![image](https://user-images.githubusercontent.com/476869/149300023-e211c08c-9e0b-4d96-be94-9661b7d4a663.png)

But I don't want to display the column. Is there a way to hide it? 

Thanks :+1:

**Comment by omeraplak** [comment]
> Hi @omeraplak
> 
> I got an error on `ReferenceError: index is not defined` from `defaultSortOrder={getDefaultSortOrder("id", index)}`. I change to `defaultSortOrder={getDefaultSortOrder("index", sorter)}` and it works!
> 
> ![image](https://user-images.githubusercontent.com/476869/149300023-e211c08c-9e0b-4d96-be94-9661b7d4a663.png)
> 
> But I don't want to display the column. Is there a way to hide it?
> 
> Thanks 👍

I also looked at how we can hide it but I guess there is no way :(

I think it's best to add the `permanentSorter` property. So it will be much easier! 
I think we can add it this week and let you know!

**Comment by niwasmala** [comment]
Thanks a lot! 

Currently I'm patching it by adding sort field to metadata and handle it on data provider.

**Comment by omeraplak** [comment]
> Thanks a lot!
> 
> Currently I'm patching it by adding sort field to metadata and handle it on data provider.

I created a PR that fixed this issue. I will be able to release when I get enough approvals. 
https://github.com/pankod/refine/pull/1430

**Comment by omeraplak** [comment]
Hey @niwasmala ,
We've released `@pankod/refine@2.6.0` with `permanentSorter` support! Could you approve this fix?
